Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Editor/EditorCommon.cpp

```diff
@@ -333,6 +333,7 @@ namespace ImageProcessingAtomEditor
         }
 
         TextureSettings& texSetting = GetMultiplatformTextureSetting();
+	AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
         for (auto& it = ++ m_settingsMap.begin(); it != m_settingsMap.end(); ++it)
```

Comment:
The temporary that is being returned from `m_settingsMap.begin()`  iterator is being assigned to a reference.

This code should just use plain `auto` without the ampersand.
```suggestion
	AZ_PUSH_DISABLE_WARNING_GCC("-Wdangling-pointer");
        for (auto it = ++ m_settingsMap.begin(); it != m_settingsMap.end(); ++it)
```